### PR TITLE
Added reporting of invalid files, test case for bad JS

### LINKED ISSUE #9: Reporting of parsing errors
Currently no errors are reported if `node-source-walk` [encounters an error while parsing a file](https://github.com/lijunle/depcheck-es6/blob/master/index.js#L51-L53), which makes it tough to differentiate between a file that simply didn't include dependencies and a file with a syntax error.

I'm not sure what the best way to report errors would be... Should it be with a debug option that triggers a `console.debug` or perhaps returns some kind of `ParsingError` object that the calling method can use... or simply re-raise the error?

### PR DESCRIPTION
**Pull request by woodb**
Closes #9

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -101,6 +101,15 @@ describe("depcheck", function () {
     });
   });
 
+  it("should report bad javascript", function testBadJS(done) {
+    var absolutePath = path.resolve("test/fake_modules/bad_js");
+
+    depcheck(absolutePath, {  }, function checked(unused) {
+      assert.notEqual(unused.invalidFiles, {});
```
Comment (lijunle) [inline test/index.js]: Why not merge this assertion into line 99 (which is commented out)?

Besides, could you add stronger assertion - use deepEqual to compare to `invalidFiles` object.

Comment (woodb) [inline test/index.js]: I thought about that, but was worried about tweaking an existing test :3 I'll add it back

Comment (woodb) [inline test/index.js]: Alright, cleaned it up in https://github.com/woodb/depcheck-es6/commit/1ac996178046b5db860f006cf09844bfb4ede2aa :+1:

File: test/index.js [hunk 2]
```diff
@@ -96,7 +96,7 @@ describe("depcheck", function () {
 
     depcheck(absolutePath, {  }, function checked(unused) {
       assert.equal(unused.dependencies.length, 1);
-      //assert.deepEqual(Object.keys(unused.invalidFiles).length, 2);
+      assert.deepEqual(Object.keys(unused.invalidFiles).length, 2);
```
Comment (lijunle) [inline test/index.js:99]: I don't think the `depcheck-es6/test/fake_modules/bad_js/deeper/also_bad_js.js` file is necessary. Please remove it and change the number to 1.

Besides, could you please add more assertion on `invalidFiles`. For example, from the `Object.keys(..).length` assertion, user cannot get the shape of the object shape of `invalidFiles` object. The test should have the ability to do so.

**Comment by woodb** [comment]
:(

**Pull request closed without merging**